src/main/java/repositories/ItemRepository.java

```diff
@@ -0,0 +1,52 @@
+package repositories;
+
+import entities.Item;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ItemRepository implements IGenericRepository<Item> {
+
+    Map<Integer, Item> items;
+
+    public ItemRepository() {
+        items = new HashMap<>();
+    }
+
+    @Override
+    public void add(Item item) {
+        items.put(item.getId(), item);
+    }
+
+    @Override
+    public Item update(Item item) {
+        if (item != null && items.containsKey(item.getId())) {
+            return items.replace(item.getId(), item);
```

Comment:
You should throw some sort of item not found exception if item is not found. Also you should return the updated item